sympy/discrete/transforms.py

```diff
@@ -36,6 +36,8 @@ def _fourier_transform(seq, dps, inverse=False):
     if n < 2:
         return a
 
+    (alpha, beta) = sympify(fourier_parameters)
```

Comment:
This should be `alpha, beta = map(as_int, fourier_parameters)` to start with.

Reply:
I think the choices may not be limited to the integers, as long as it can be used as exponents.

Reply:
When can these parameters be non-integral?

Reply:
For the second parameter, even if set to integer values, I think some values like `0` or such can break the fft scheme, because the lower right block of the matrix would be positive copy of the upper right block, and also inverse fourier may not restore the original numbers.

Also, some other integers can bring some effect like above, and things can get more complicated.

I may restrict the second parameter to 1 or -1 strictly.

But for the first one, I don't know the exact consequences of using non-integers can go wrong